main.py

```diff
@@ -27,14 +27,14 @@ class MainPage(webapp2.RequestHandler):
   def get(self):
     try:
       template = find_template(self.request.path)
-      self.response.headers['Cache-Control'] = 'public, max-age=60'
-      self.response.write(template.render({}))
     except jinja2.exceptions.TemplateNotFound:
+      template = find_template('/404.html')
       self.response.set_status(404)
-      self.response.write('Not found')
     except Exception:
+      template = find_template('/500.html')
       self.response.set_status(500)
-      self.response.write('Server error')
+    self.response.headers['Cache-Control'] = 'public, max-age=60'
```

Comment:
Don't cache error responses

Reply:
Good call. Fixed.

Reply:
Sorry wasn't clear - meant to say don't cache 404 _and_ 500 responses. I can fix this before merging.

Reply:
No worries. Not caching 404 response now.